### DISCUSSION TITLE
✅ Ability to launch a command (or snippet) on one or multiple hosts

### DISCUSSION THREAD

**Discussion by Dag0d** [isontheline/pro.webssh.net#560] [Ideas]
I’m using WebSSH on a nearly daily basis to manage my home lab across all my devices, I love it.
During my use I’ve notices a few things I would change/add.

The first thing would be an auto password fill.
When ever I type sudo XX and I’m asked for my password, I would love to see a special key which will enter the saved password for this host.
Currently my workaround is, to have a snippet for each host but this is not ideal in my opinion 

Talking about snippets, it would be nice to add snippets based on hosts, like having global snippets and host specific snippets. Same applies to the “preferred” option

Another nice feature would be multi host snippets execution.

For the SFTP browser I would really like to see a more “iOS-/MacOS-ish” interface maybe with a side by side window to access local file. 

I also miss the option to download folders.

I would be more than happy to help you with all those requests.
Thanks for this amazing part of software.

**Comment by isontheline** [comment]
Hello @Dag0d 👋

> I’m using WebSSH on a nearly daily basis to manage my home lab across all my devices, I love it. 

Thank you a lot for using WebSSH and supporting it 👍

> During my use I’ve notices a few things I would change/add.
> The first thing would be an auto password fill. When ever I type sudo XX and I’m asked for my password, I would love to see a special key which will enter the saved password for this host. Currently my workaround is, to have a snippet for each host but this is not ideal in my opinion

It's a nice feature!
Are you agree with this statement : "Ability to paste the current connection password by pressing a key"?
If so I will create an issue to track it's implementation.

> Talking about snippets, it would be nice to add snippets based on hosts, like having global snippets and host specific snippets. Same applies to the “preferred” option

This is already possible : https://webssh.net/documentation/help/howtos/link-connections-and-snippets/
* Does this how-to helps you?
* Does the documentation need to be rewritten?

> Another nice feature would be multi host snippets execution.

Nice one too.
Are you agree with this statement : "Ability to launch a snippet on multiple hosts. No terminals will be launched, only results will be displayed"?

> For the SFTP browser I would really like to see a more “iOS-/MacOS-ish” interface maybe with a side by side window to access local file.

I will try to implement local file access. 
Currently WebSSH is in a sandbox (App Store' rule).

> I also miss the option to download folders.

I think it's related to : https://github.com/isontheline/pro.webssh.net/issues/205

> I would be more than happy to help you with all those requests. Thanks for this amazing part of software.

Thank you too for your strong feedback 🙏

**Reply by Dag0d** [reply]
Hi @isontheline,

thanks for the prompt reply!

I agree with both of your statements, I think both will be great enhancements.

My bad, I must have missed the tag part for snippets completely, sorry for that.
The documentation helped, thanks.

Even in the sandbox you should be able to use the file picker from apple.

you are right #205 is exactly what I’ve been talking about.

many thanks!

**Reply by isontheline** [reply]
You're Welcome @Dag0d 

I have created two issues in order to track their progress :
[Ability to paste the current connection password by pressing a key](https://github.com/isontheline/pro.webssh.net/issues/563)
[Ability to launch a snippet on multiple hosts](https://github.com/isontheline/pro.webssh.net/issues/564)

**Comment by isontheline** [comment]
I'm working on #564 now
